### PR TITLE
Fixed time datatype conversion in salesforce sink

### PR DESCRIPTION
**Pull request by Aryan-Verma**
Salesforce Sink plugin did not support Time datatype due to wrong conversion to local time. [PLUGIN-1631](https://cdap.atlassian.net/browse/PLUGIN-1631)

Also fixed unsupported Error Handling property 'Fail on Error': [PLUGIN-1632](https://cdap.atlassian.net/browse/PLUGIN-1632)

[PLUGIN-1631]: https://cdap.atlassian.net/browse/PLUGIN-1631?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[PLUGIN-1632]: https://cdap.atlassian.net/browse/PLUGIN-1632?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/StructuredRecordToCSVRecordTransformer.java [hunk 1]
```diff
@@ -63,7 +65,8 @@ public static String convertSchemaFieldToString(Object value, Schema.Field field
         case TIME_MICROS:
           // convert timestamp to HH:mm:ss,SSS
           instant = Instant.ofEpochMilli(TimeUnit.MICROSECONDS.toMillis((Long) value));
-          return instant.atZone(ZoneOffset.UTC).toLocalTime().toString();
+          DateTimeFormatter timeFormat = DateTimeFormatter.ofPattern("HH:mm:ss.SSS'Z'");
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/StructuredRecordToCSVRecordTransformer.java:68]: Please add a unit test for this.

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/ErrorHandling.java [hunk 2]
```diff
@@ -25,7 +25,7 @@ public enum ErrorHandling {
 
   SKIP("Skip on error"),
 
-  STOP("Stop on error");
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/ErrorHandling.java:28]: Looks like this value is being used in some tests - https://github.com/data-integrations/salesforce/blob/cffa7e499e6ea80ff290640c062852c5d3f00942/src/test/java/io/cdap/plugin/salesforce/etl/BaseSalesforceBatchSinkETLTest.java#L191 . Can this be fixed and a unit test be added with the new type?

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/ErrorHandling.java [hunk 3]
```diff
@@ -25,7 +25,7 @@ public enum ErrorHandling {
...
+  FAIL("Fail on error");
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/ErrorHandling.java:28]: Where is this being used? If there are existing references, they should reflect the change.

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/StructuredRecordToCSVRecordTransformer.java (hunk 1 above)
Comment (Aryan-Verma) [reply src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/StructuredRecordToCSVRecordTransformer.java:68]: Added

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/ErrorHandling.java (hunk 2 above)
Comment (Aryan-Verma) [reply src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/ErrorHandling.java:28]: Fixed this

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/ErrorHandling.java (hunk 3 above)
Comment (MrRahulSharma) [reply src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/ErrorHandling.java:28]: ErrorHandling.STOP is not used anywhere in the code. The callers are rather using the methd ErrorHandling.fromValue(string) method call instead. The value in the widget json("Fail on error") does not match with the Enum string ("Stop on error") which currently cause the validate operation to fail in the method : https://github.com/data-integrations/salesforce/blob/cffa7e499e6ea80ff290640c062852c5d3f00942/src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceSinkConfig.java#L223

File: src/test/java/io/cdap/plugin/salesforce/plugin/sink/batch/StructuredRecordToCSVRecordTransformerTest.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
```
Comment (greeshmaswaminathan) [inline src/test/java/io/cdap/plugin/salesforce/plugin/sink/batch/StructuredRecordToCSVRecordTransformerTest.java]: Change to 2023

File: src/test/java/io/cdap/plugin/salesforce/plugin/sink/batch/StructuredRecordToCSVRecordTransformerTest.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+ * the License.
+ */
+package io.cdap.plugin.salesforce.plugin.sink.batch;
+
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class StructuredRecordToCSVRecordTransformerTest {
```
Comment (greeshmaswaminathan) [inline src/test/java/io/cdap/plugin/salesforce/plugin/sink/batch/StructuredRecordToCSVRecordTransformerTest.java:23]: Please add java doc.

**Review by greeshmaswaminathan — APPROVED** [review]
A couple of minor comments , otherwise lgtm. Please squash the commits after the fix.

File: src/test/java/io/cdap/plugin/salesforce/plugin/sink/batch/StructuredRecordToCSVRecordTransformerTest.java (hunk 4 above)
Comment (Aryan-Verma) [reply src/test/java/io/cdap/plugin/salesforce/plugin/sink/batch/StructuredRecordToCSVRecordTransformerTest.java]: Done

File: src/test/java/io/cdap/plugin/salesforce/plugin/sink/batch/StructuredRecordToCSVRecordTransformerTest.java (hunk 5 above)
Comment (Aryan-Verma) [reply src/test/java/io/cdap/plugin/salesforce/plugin/sink/batch/StructuredRecordToCSVRecordTransformerTest.java:23]: Done

**Review by MrRahulSharma — APPROVED** [review]

**Pull request merged**